subprojects/packagefiles/openssl/generator.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+
+# Change to the directory of the Bash script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+os.chdir(script_dir)
+
+# Node.js version should bundle OpenSSL of matching version to one specified in wrap file
+node_version = "v19.7.0"
+openssl_version = os.environ.get("OPENSSL_VERSION")
+
+if openssl_version is None:
+    with open("../../openssl.wrap") as wrap_file:
+        for line in wrap_file:
+            if "directory = " in line:
+                openssl_version = line.strip().split()[-1]
+                break
+
+# Clone the Node.js repository
+subprocess.run(["git", "clone", "--depth", "1", "-b", node_version, "https://github.com/nodejs/node.git"])
+
+# Remove the generated-config directory
+subprocess.run(["rm", "-rf", "generated-config"])
+
+# Change directory to Node.js deps/openssl
+os.chdir("node/deps/openssl")
+
+# Apply the patch to generate meson.build for different targets
+subprocess.run(["patch", "-u", "config/generate_gypi.pl", "-i", "../../../generate_gypi.pl.patch"])
+
+# Copy meson.build template file
+subprocess.run(["cp", "../../../meson.build.tmpl", "config/"])
+
+# Clone the upstream OpenSSL repository
+subprocess.run(["rm", "-rf", "openssl"])
+subprocess.run(["git", "clone", "--depth", "1", "--branch", "openssl-" + openssl_version, "https://github.com/openssl/openssl.git"])
+
+# Remove the config/archs directory
+subprocess.run(["rm", "-rf", "config/archs"])
+
+# Run the make command with LANG set to C
+os.environ['LANG'] = 'C'
+subprocess.run(["make", "-C", "config"])
+
+# Copy generated files back into the correct place
+find_cmd = 'mkdir -p ../../../generated-$(dirname "$1"); cp "$1" ../../../generated-"$1"'
+subprocess.run(["find", "config/archs", "-name", "meson.build", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-name", "*.asm", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-name", "*.c", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-name", "*.h", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-iname", "*.s", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
```

Comment:
I still do not understand the purpose of this PR.

But this right here.

This is not rewriting the generator from bash into python. This is just... no.

Reply:
A proper rewrite would be much more complex to deal with globbing and whatnot.

Reply:
Ok?

So what exactly is wrong with the bash version.

Reply:
Alpine CI is lightweight and does not come with bash by default. It's probably not a good idea to mandate bash.

Reply:
Install a normal userland on the alpine CI then? idk what the problem is.

bash is more lightweight than python, that's for sure.

Reply:
but python is needed for meson anyway.

Reply:
@eli-schwartz updated version and ported more of these calls to python.

Reply:
There's still a bunch of find -execs in there...

Reply:
I came up with

```
+def copy_files(extension):
+    for root, _, files in os.walk("config/archs"):
+        for file in files:
+            if file.lower().endswith(extension.lower()):
+                source_path = os.path.join(root, file)
+                relative_path = os.path.relpath(source_path, search_dir)
+                destination_dir = os.path.join("../../../generated", os.path.dirname(relative_path))
+                destination_path = os.path.join(destination_dir, file)
+
+                os.makedirs(destination_dir, exist_ok=True)
+                shutil.copy(source_path, destination_path)
+
 # Change to the directory of the Bash script
 script_dir = os.path.dirname(os.path.abspath(__file__))
 os.chdir(script_dir)
@@ -47,12 +59,11 @@ os.environ['LANG'] = 'C'
 subprocess.run(["make", "-C", "config"])

 # Copy generated files back into the correct place
-find_cmd = 'mkdir -p ../../../generated-$(dirname "$1"); cp "$1" ../../../generated-"$1"'
-subprocess.run(["find", "config/archs", "-name", "meson.build", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
-subprocess.run(["find", "config/archs", "-name", "*.asm", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
-subprocess.run(["find", "config/archs", "-name", "*.c", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
-subprocess.run(["find", "config/archs", "-name", "*.h", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
-subprocess.run(["find", "config/archs", "-iname", "*.s", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+copy_files("meson.build")
+copy_files(".asm")
+copy_files(".c")
+copy_files(".h")
+copy_files(".s")
```

I don't think it's equivalent.